zrml/prediction-markets/src/benchmarks.rs

```diff
@@ -215,18 +198,72 @@ fn setup_resolve_common_scalar<T: Config>(
     Ok((caller, market_id))
 }
 
-// Setup a disputed scalar market and create accounts with outcome assets.
-fn setup_resolve_common_scalar_after_dispute<T: Config>(
-    acc_total: u32,
-    acc_asset: u32,
+fn setup_reported_categorical_market_with_pool<T: Config>(
+    categories: u32,
 ) -> Result<(T::AccountId, MarketIdOf<T>), &'static str> {
-    let (caller, market_id) = setup_resolve_common_scalar::<T>(acc_total, acc_asset)?;
-    Pallet::<T>::dispute(
+    let (caller, market_id) = create_market_common::<T>(
+        MarketCreation::Permissionless,
+        MarketType::Categorical(categories.saturated_into()),
+        ScoringRule::CPMM,
+        None,
+    )?;
+
+    let max_swap_fee: BalanceOf<T> = MaxSwapFee::get().saturated_into();
+    let min_liquidity: BalanceOf<T> = MinLiquidity::get().saturated_into();
+    let _ = Call::<T>::buy_complete_set { market_id, amount: min_liquidity }
+        .dispatch_bypass_filter(RawOrigin::Signed(caller.clone()).into())?;
+    let weight_len: usize = MaxRuntimeUsize::from(categories).into();
+    let weights = vec![MinWeight::get(); weight_len];
+    Pallet::<T>::deploy_swap_pool_for_market(
+        RawOrigin::Signed(caller.clone()).into(),
+        market_id,
+        max_swap_fee,
+        min_liquidity,
+        weights,
+    )
+    .unwrap();
```

Comment:
You made a great point in some other PR that our helper functions are returning results, so we shouldn't use `unwrap()` here, and use the question mark operator instead.

Reply:
So error is batter than painc right? as it can prpogate error from its origin, for example in this if something goes wrong in deploy_swap...() using ? is better than just unwrap() which may point that line 217 fails :thinking: